cube/cube.cpp

```diff
@@ -193,27 +193,29 @@ static const float g_uv_buffer_data[] = {
 };
 // clang-format on
 
-typedef struct {
+struct SwapchainImageResources {
     vk::Image image;
     vk::CommandBuffer cmd;
     vk::CommandBuffer graphics_to_present_cmd;
     vk::ImageView view;
     vk::Buffer uniform_buffer;
     vk::DeviceMemory uniform_memory;
-    void *uniform_memory_ptr;
+    void *uniform_memory_ptr = nullptr;
     vk::Framebuffer framebuffer;
     vk::DescriptorSet descriptor_set;
-} SwapchainImageResources;
+};
 
 struct Demo {
     Demo();
-    void build_image_ownership_cmd(uint32_t const &);
-    vk::Bool32 check_layers(uint32_t, const char *const *, uint32_t, vk::LayerProperties *);
+    void build_image_ownership_cmd(const SwapchainImageResources &);
+    vk::Bool32 check_layers(const std::vector<const char *> &check_names, const std::vector<vk::LayerProperties> &layers);
```

Comment:
Some of the function signatures use variable names (this one, `destroy_texture()`, `draw_build_cmd()`, others).  Some do not (`init()`, `prepare_shader_module()`).  They should all match, right?

Reply:
fixed.